lib/internal/modules/esm/loader.js

```diff
@@ -158,27 +158,29 @@ class ESMLoader {
   async getModuleJob(specifier, parentURL, importAssertions) {
     let importAssertionsForResolve;
 
-    // We can skip cloning if there are no user-provided loaders because
-    // the Node.js default resolve hook does not use import assertions.
     if (this.#hooks?.hasCustomLoadHooks) {
       importAssertionsForResolve = {
         __proto__: null,
         ...importAssertions,
       };
+    } else {
+      // We can skip cloning if there are no user-provided loaders.
+      importAssertionsForResolve = importAssertions;
```

Comment:
We can keep it undefined, by spec the default resolver should not be impacted by import assertions, so let's not give it to it.
```suggestion
```

Reply:
We need to pass the assertions into default resolve so that it can return them when a user resolve hook calls `next`.

Reply:
I don't think it's a good idea, IMO the default resolve should stick to ignoring import assertions